packages/jest-circus/src/index.js

```diff
@@ -46,13 +46,20 @@ const _dispatchDescribe = (
     asyncError.message = `Invalid second argument, ${blockFn}. It must be a callback function.`;
     throw asyncError;
   }
+
   dispatch({
     asyncError,
     blockName,
     mode,
     name: 'start_describe_definition',
   });
-  blockFn();
+  const describeReturn = blockFn();
+  if (isPromise(describeReturn)) {
```

Comment:
we don't actually care if it's a promise or not - the return type should be `void` (`undefined`). Whether it's a promise or not we don't care about (although the promise case is more probable to give a runtime error).

We already yell at you for returning anything not a promise from `it/test` - this is the same except we don't accept promises either

Reply:
Thought about forbidding anything but `undefined` too, but then it gets *really* breaking IMO. I'm sure *someone* has something like
```js
describe('thing', () => {
  return it('does stuff', () => { });
}
```
in their suite

Reply:
Forbid promises now, anything in Jest 25?

Reply:
print deprecation warning now without throwing, then throw for real in 25?

/cc @thymikee @rubennorte

Reply:
that example wouldn't throw btw - `it` returns `undefined`

Reply:
You wish 😄 
![image](https://user-images.githubusercontent.com/16069751/52535315-23ed5980-2d4d-11e9-972e-4db4b3ad00ca.png)

Reply:
hah. we should make sure to return `undefined` then

Reply:
That would be another non-breaking one we could do now in 24 the way I see it

Reply:
It's safer to log a warning now and make it work properly in next major

Reply:
Do we have any infrastructure in place to emit warnings from `circus/jasmine2`? Not sure how I'd get one through to the main process

Reply:
you can use warnings from jest-validate

Reply:
That's pretty much just `console.warn` as well though - I just checked, for jasmine2 it's fine to print a warning to the console, but for circus the warnings won't show up

Reply:
Didn't find anything appropriate in `jest-validate` (mostly option validation stuff) so I went with plain `console.warn`